nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java

```diff
@@ -195,14 +196,27 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
       if (isSnapshotOperation(base, metadata)) {
         builder.putProperties("iceberg.operation", snapshot.operation());
       }
-      Branch branch =
+
+      CommitMultipleOperationsBuilder commit =
           client
               .getApi()
               .commitMultipleOperations()
               .operation(Operation.Put.of(key, newTable, table))
               .commitMeta(NessieUtil.catalogOptions(builder, catalogOptions).build())
-              .branch(expectedHead)
-              .commit();
+              .branch(expectedHead);
+
+      if (base == null) { // new table
+        client
+            .findImpliedNamespaces(key)
+            .forEach(
+                ns -> {
+                  LOG.info(
+                      "Adding implied Namespace '{}' in '{}'", ns, client.getRef().getReference());
+                  commit.operation(Operation.Put.of(ContentKey.of(ns.getElements()), ns));
```

Comment:
nit: does keeping namespace put operation before the table put operation makes sense?

Reply:
I believe the order of operations inside a commit does not matter conceptually.